File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsPeerReview.swift
import Foundation

enum PeerReviewAnonymity: String, Codable, Hashable {
    case doubleBlind = "double_blind"
    case reviewerAnon = "reviewer_anon"
    case named
}

enum PeerReviewAllocationStatus: String, Codable, Hashable {
    case assigned
    case inProgress = "in_progress"
    case submitted
    case expired
}

struct PeerReviewAllocation: Codable, Identifiable, Hashable {
    var id: String
    var configId: String
    var assignmentId: String
    var courseId: String
    var courseCode: String
    var targetSubmissionId: String
    var status: PeerReviewAllocationStatus
    var assignedAt: String
    var anonymity: PeerReviewAnonymity
    var targetLabel: String?
    var targetUserId: String?
    var closesAt: String?
    var opensAt: String?
}

struct PeerReviewAssignedResponse: Decodable {
    var allocations: [PeerReviewAllocation]
}

struct PeerReviewReceivedItem: Codable, Identifiable, Hashable {
    var id: String
    var score: Double?
    var comments: String?
    var submittedAt: String
    var reviewerLabel: String?
}

struct PeerReviewReceivedResponse: Decodable {
    var reviews: [PeerReviewReceivedItem]
}

struct PeerReviewExistingReview: Codable, Hashable {
    var id: String
    var allocationId: String
    var score: Double?
    var rubricScores: [String: Double]?
    var comments: String?
    var submittedAt: String
}

struct PeerReviewAllocationDetail: Decodable {
    var allocation: PeerReviewAllocation
    var assignmentTitle: String?
    var submission: AssignmentSubmission
    var rubric: RubricDefinition?
    var review: PeerReviewExistingReview?
}

struct PeerReviewSubmitRequest: Encodable {
    var score: Double?
    var rubricScores: [String: Double]?
    var comments: String?
}

struct PeerReviewSubmitResponse: Decodable {
    var id: String
    var allocationId: String
    var score: Double?
    var submittedAt: String
}

struct PeerReviewListRoute: Hashable {}

struct PeerReviewDetailRoute: Hashable {
    var allocationId: String
}

struct PeerReviewsReceivedRoute: Hashable {
    var courseCode: String
    var assignmentId: String
    var assignmentTitle: String
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsPeerReview.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
enum class PeerReviewAnonymity {
    @Suppress("unused") double_blind,
    @Suppress("unused") reviewer_anon,
    @Suppress("unused") named,
}

@Serializable
enum class PeerReviewAllocationStatus {
    @Suppress("unused") assigned,
    @Suppress("unused") in_progress,
    @Suppress("unused") submitted,
    @Suppress("unused") expired,
}

@Serializable
data class PeerReviewAllocation(
    val id: String,
    val configId: String,
    val assignmentId: String,
    val courseId: String,
    val courseCode: String,
    val targetSubmissionId: String,
    val status: PeerReviewAllocationStatus,
    val assignedAt: String,
    val anonymity: PeerReviewAnonymity,
    val targetLabel: String? = null,
    val targetUserId: String? = null,
    val closesAt: String? = null,
    val opensAt: String? = null,
)

@Serializable
data class PeerReviewAssignedResponse(
    val allocations: List<PeerReviewAllocation> = emptyList(),
)

@Serializable
data class PeerReviewReceivedItem(
    val id: String,
    val score: Double? = null,
    val comments: String? = null,
    val submittedAt: String,
    val reviewerLabel: String? = null,
)

@Serializable
data class PeerReviewReceivedResponse(
    val reviews: List<PeerReviewReceivedItem> = emptyList(),
)

@Serializable
data class PeerReviewExistingReview(
    val id: String,
    val allocationId: String,
    val score: Double? = null,
    val rubricScores: Map<String, Double>? = null,
    val comments: String? = null,
    val submittedAt: String,
)

@Serializable
data class PeerReviewAllocationDetail(
    val allocation: PeerReviewAllocation,
    val assignmentTitle: String? = null,
    val submission: AssignmentSubmission,
    val rubric: RubricDefinition? = null,
    val review: PeerReviewExistingReview? = null,
)

@Serializable
data class PeerReviewSubmitRequest(
    val score: Double? = null,
    val rubricScores: Map<String, Double>? = null,
    val comments: String? = null,
)

@Serializable
data class PeerReviewSubmitResponse(
    val id: String,
    val allocationId: String,
    val score: Double? = null,
    val submittedAt: String,
)
